src/components/WarningModal/WarningModal.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  Text,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+} from "@chakra-ui/react"
+import { ModalCloseButton } from "@opengovsg/design-system-react"
+import { PropsWithChildren } from "react"
+
+interface ModalProps {
```

Comment:
we probably want to extend `ModalProps` from chakra and call this `WarningModalProps` so this should look something like: 

```ts
interface WarningModalProps extends ModalProps { 
displayTitle: string 
displayText: JSX.Element
}
```

also, could you verify if strings/`undefined` etc is assignable to `JSX.Element`? they should be as consumers might wish to pass only a string/don't pass anything. 

then we can just use `...rest` and spread it on the modal itself.

Reply:
Makes sense, fixed in https://github.com/isomerpages/isomercms-frontend/pull/955/commits/ac3d56c66dedd4ff46f524f52c8ef05d865538bd

Strings/undefined values seem to be fine!